### PR TITLE
1 oauth

### PR DESCRIPTION
**Pull request by jozefvaclavik**

### CODE REVIEW

**Comment by ryanwjackson** [comment]
@jozefvaclavik I was trying to get this server running using Convox (what we use to run and deploy apps).  Here is the error I get:

```
build  │ Step 9/10 : RUN bundle exec rake assets:precompile
build  │  ---> Running in ed3654f974ef
build  │ rake aborted!
build  │ NameError: uninitialized constant OmniAuth::Strategies::PaidConnect
build  │ /app/config/initializers/devise.rb:255:in `block in <top (required)>'
build  │ /var/lib/gems/2.3.0/gems/devise-4.2.1/lib/devise.rb:296:in `setup'
build  │ /app/config/initializers/devise.rb:3:in `<top (required)>'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/engine.rb:648:in `block in load_config_initializer'
build  │ /var/lib/gems/2.3.0/gems/activesupport-5.0.2/lib/active_support/notifications.rb:166:in `instrument'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/engine.rb:647:in `load_config_initializer'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/engine.rb:612:in `block (2 levels) in <class:Engine>'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/engine.rb:611:in `each'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/engine.rb:611:in `block in <class:Engine>'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/initializable.rb:30:in `instance_exec'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/initializable.rb:30:in `run'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/initializable.rb:55:in `block in run_initializers'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/initializable.rb:44:in `each'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/initializable.rb:44:in `tsort_each_child'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/initializable.rb:54:in `run_initializers'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/application.rb:352:in `initialize!'
build  │ /app/config/environment.rb:5:in `<top (required)>'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/application.rb:328:in `require_environment!'
build  │ /var/lib/gems/2.3.0/gems/railties-5.0.2/lib/rails/application.rb:448:in `block in run_tasks_blocks'
build  │ /var/lib/gems/2.3.0/gems/sprockets-rails-3.2.0/lib/sprockets/rails/task.rb:62:in `block (2 levels) in define'
build  │ /var/lib/gems/2.3.0/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
build  │ Tasks: TOP => environment
build  │ (See full trace by running task with --trace)
build  │ The command '/bin/sh -c bundle exec rake assets:precompile' returned a non-zero code: 1
```

I suspect the `rake assets:precompile` is not resulting in the `lib/...` to be autoloaded, so `OmniAuth::Strategies::PaidConnect` has not been loaded by the time the Devise initializer runs.  Does that make sense?

If you have Convox CLI installed, you can go into the root of that directory and run `convox init` and then `convox start`.  It may require some more configuration, but this was as far as I got.

## Update

This is resolved by removing the `bundle exec rake assets:precompile` which isn't the solution, but it can get it running locally.  I assume you can reproduce by just running the precompile command.

**Comment by ryanwjackson** [comment]
Otherwise, everything works.  Also, if you want, you can check out the #2 Convox Branch/PR.  You run with `convox start`.

**Review by ryanwjackson — CHANGES_REQUESTED** [review]
Please ensure precompile works before we sign off.

Also, can you write clear instructions on how to run.  I imagine something like:
1. Create .env in root
2. Add the following:
```
PAID_APP_ID=...
PAID_SECRET=...
```
3. Run `bundle exec rails server`
4. Update your Paid application redirect URI to be ...

File: app/lib/omni_auth/strategies/paid_connect.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# encoding: UTF-8
+module OmniAuth
+  module Strategies
+    # PaidConnect strategy for Omniauth
+    class PaidConnect < OmniAuth::Strategies::OAuth2
+      option :name, :paid_connect
+
+      option :client_options, site: 'https://auth.paidlabs.com'
```
Comment (amedrz) [inline app/lib/omni_auth/strategies/paid_connect.rb]: what if we use ENV variables for this kind of things? Just as we already do with `PAID_APP_ID` etc.

Comment (ryanwjackson) [inline app/lib/omni_auth/strategies/paid_connect.rb]: 👍

Comment (ryanwjackson) [reply app/lib/omni_auth/strategies/paid_connect.rb]: Maybe something like `site: ENV['PAID_OAUTH_URI'] || 'https://auth.paidlabs.com'`

Comment (amedrz) [reply app/lib/omni_auth/strategies/paid_connect.rb]: yeah, actually it's recommended to use `.fetch` instead: `ENV.fetch('PAID_OAUTH_URI', 'https://auth.paidlabs.com')`

**Review by ryanwjackson — APPROVED** [review]

**Pull request merged**